posthog/middleware.py

```diff
@@ -370,3 +370,53 @@ def __call__(self, request: HttpRequest):
 
         response = self.get_response(request)
         return response
+
+
+def per_request_logging_context_middleware(
+    get_response: Callable[[HttpRequest], HttpResponse]
+) -> Callable[[HttpRequest], HttpResponse]:
+    """
+    We get some default logging context from the django-structlog middleware,
+    see
+    https://django-structlog.readthedocs.io/en/latest/getting_started.html#extending-request-log-metadata
+    for details. They include e.g. request_id, user_id. In some cases e.g. we
+    add the team_id to the context like the get_events and decide endpoints.
+
+    This middleware adds some additional context at the beggining of the
+    request. Feel free to add anything that's relevant for the request here.
+    """
+
+    def middleware(request: HttpRequest) -> HttpResponse:
+        # Add in the host header, and the x-forwarded-for header if it exists.
+        # We add these such that we can see if there are any requests on cloud
+        # that do not use Host header app.posthog.com. This is important as we
+        # roll out CloudFront in front of app.posthog.com. We can get the host
+        # header from NGINX, but we really want to have a way to get to the
+        # team_id given a host header, and we can't do that with NGINX.
+        structlog.contextvars.bind_contextvars(
+            host=request.META.get("HTTP_HOST", ""),
+            x_forwarded_for=request.META.get("HTTP_X_FORWARDED_FOR", ""),
+        )
+
+        return get_response(request)
+
+    return middleware
+
+
+def user_logging_context_middleware(
+    get_response: Callable[[HttpRequest], HttpResponse]
+) -> Callable[[HttpRequest], HttpResponse]:
+    """
+    This middleware adds the team_id to the logging context if it exists. Note
+    that this should be added after we have performed authentication, as we
+    need the user to be authenticated to get the team_id.
+    """
+
+    def middleware(request: HttpRequest) -> HttpResponse:
+        if request.user.is_authenticated:
```

Comment:
I remember we had some problems with the rate limiting code from assuming team id was present...

So, we could be more cautious reading the user and its team...

But non-blocking - I might be bike-shedding

Reply:
Hah good catch 🙏 

I hope the tests caught something if it's an issue, will check.

Reply:
Do you happen to have a link to the limiting PR that had the issue?

Reply:
It lead to this method https://github.com/PostHog/posthog/blob/d196ea53cc7971c6ed1f7cff1fdae0682fe2c861/posthog/rate_limit.py#L53

Reply:
Ah, it was _views_ that don't have a team id https://github.com/PostHog/posthog/pull/12006